### PR TITLE
Verify addresses in hex files

### PR DESCRIPTION
**Pull request by hydra**
Verify target has all addresses found in the hex file instead of verifying the size.

As noted in PR #2512 simply verifying the size of the hex file is not extensible and is not how hex files are designed to be used.

A hex file can contains many non-contiguous blocks of data.

This also supports targets with any partition layout.

Additionally, this provides a more future-proof and flexible way of providing hex files which can now contain data for non-contiguous blocks, e.g. firmware + config, or firmware + osd fonts.

This replaces PR #2512

Tested on:

- SPRacingH7RF (External Flash, Different partition layout from previous FCs, image file matches partition size)
```@External Flash /0x90000000/111*8Kg,16*8Kg,1*8Kg,128*8Kg,0*8Ka```
- SPRacingH7EXTREME (External Flash, Partition larger than .hex file - Partition: 512KB, Hex: 448KB)
```"@External Flash /0x90000000/998*128Kg,1*128Kg,4*128Kg,21*128Ka```
- SPRacingF7DUAL (Internal Flash, Unified target, .hex spans multiple sectors)
```@Internal Flash  /0x08000000/04*016Kg,01*64Kg,03*128Kg```

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by McGiverGim — COMMENTED** [review]
You don't need to remove the message at each languages file. They will be removed (or replaced) when we update the languages from Crowdin if the master (English) has changed.
I understand that in the nightlies it will appear "wrong" until this happens, but the PR will be a lot smaller in changes.
I can do a PR with the updates of the languages in some days, if needed.

**Comment by hydra** [comment]
> You don't need to remove the message at each languages file. They will be removed (or replaced) when we update the languages from Crowdin if the master (English) has changed.
> I understand that in the nightlies it will appear "wrong" until this happens, but the PR will be a lot smaller in changes.
> I can do a PR with the updates of the languages in some days, if needed.

i thought so but wasn't sure which is why i did that in a separate commit.  i can remove the commit if you like?

**Comment by schugabe** [comment]
I tested to flash a SPRacingH7RF (selected hex file manually) and a CRAZYBEEF4FR with 4.2.9 using this new PR and it worked without any issues.

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by haslinghuis** [comment]
My board (SPRacing H750 Extreme) should arrive soon so I'll be able to test

**Comment by hydra** [comment]
@haslinghuis I'm not sure there's much value waiting for a board that I've already tested this with.  If you want to test with existing targets that might be useful, but I did that too. (F7DUAL).  @schugabe also tested with the CRAZYBEEF4FR and SPRACINGH7RF too.

**Review by haslinghuis — DISMISSED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by hydra** [comment]
Rebased on master.

File: src/js/protocols/stm32usbdfu.js
Comment (haslinghuis) [inline src/js/protocols/stm32usbdfu.js:612]: Why remove this exception?

Comment (hydra) [reply src/js/protocols/stm32usbdfu.js:612]: that's odd, I don't recall doing that, well spotted, I will re-instate.

Comment (hydra) [reply src/js/protocols/stm32usbdfu.js:612]: Ahh, I see what happened, it moved upwards to line 619, see above.

**Review by haslinghuis — APPROVED** [review]

**Comment by hydra** [comment]
@haslinghuis what's stopping this being merged?

**Comment by Asizon** [comment]
@hydra you need to remove all language translations, we just need english change.

**Comment by hydra** [comment]
@Asizon @haslinghuis Done. I removed the commit that deleted the old translations, no other changes.

**Review by limonspb — APPROVED** [review]

**Review by blckmn — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (PASS)
 - github identifies PR as mergeable -> PASS
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> PASS

**Pull request merged**
